src/prefect/results.py

```diff
@@ -511,6 +526,10 @@ async def create(
     class Config:
         extra = "forbid"
 
+    @classmethod
+    def __dispatch_key__(cls, **kwargs):
+        return cls.__fields__.get("type").get_default()
```

Comment:
I will swap `__fields__` -> `model_fields` once I have these classes inherit from the compat `BaseModel`